x-pack/plugins/observability_solution/apm/ftr_e2e/cypress/e2e/service_inventory/service_inventory.cy.ts

```diff
@@ -88,7 +88,7 @@ describe('Service inventory', () => {
       cy.visitKibana(serviceInventoryHref);
     });
 
-    it('with the correct environment when changing the environment', () => {
+    it.skip('with the correct environment when changing the environment', () => {
```

Comment:
@michaelolo24 these tests are not failing on main AFAICT

Reply:
I unskipped all the tests, let's see what the build says!

Reply:
@cauemarcondes over the last 2 builds ([this one](https://buildkite.com/elastic/kibana-pull-request/builds/251365#01932b38-61bd-4715-89d0-72811c1d1031) and [that one](https://buildkite.com/elastic/kibana-pull-request/builds/251045#01932745-c96b-4783-9e3b-bc9f4a0959f3)) we have 3 tests failing:
- `service_inventory/service_inventory.cy.ts`
- `service_overview/service_overview.cy.ts`
- `settings/agent_configurations.cy.ts`

These are the 3 tests that @michaelolo24 has skipped and I unskipped in my [last commit](https://github.com/elastic/kibana/pull/198715/commits/8efa7e3a647893b259412619ced56bd473b7a5bf).
If these are not failing on `main`, can you help us figuring out why these 3 tests would consistently fail?

Reply:
@PhilippeOberti / @michaelolo24 after checking out your branch and running the test I could see any specific problem with the tests you mentioned. They are breaking as CI pointed but there is no clear reason why. This is the error showed on the console:
```
2024-11-18 11:58:49.047 Cypress Helper (Renderer)[30120:5226514] CoreText note: Set a breakpoint on CTFontLogSystemFontNameRequest to debug.

We detected that the Electron Renderer process just crashed.

We have failed the current spec but will continue running the next spec.

This can happen for a number of different reasons.

If you're running lots of tests on a memory intense application.
  - Try increasing the CPU/memory on the machine you're running on.
  - Try enabling experimentalMemoryManagement in your config file.
  - Try lowering numTestsKeptInMemory in your config file during 'cypress open'.

You can learn more here:

https://on.cypress.io/renderer-process-crashed

  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     11 seconds                                                                       │
  │ Spec Ran:     service_inventory.cy.ts                                                          │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

I double-cheked and the tests are indeed passing on the main branch.

Reply:
Every test that requires deleting the current text in the `data-test-subj="environmentFilter"` and `data-test-subj="serviceNameComboBox"` fails. For whatever reason when cypress attempts to take that action, cypress just crashes, but it makes sense based on my interaction with the component in the UI above.

Besides that, the integration_policy and tutorial files didn't load at all for whatever reason when I tested last week.

Reply:
>You can see in the video below, that even through the UI, there is no way to actually clear those fields

I wonder how the tests are passing on the main branch though. That's my point actually. This code is 2-years-old and it's not flaky. Thus something that was added on this PR make it fail.

Reply:
Cypress might just be behaving closer to an actual user interaction on this newer version rather than overriding the changes. I can take a look and see if any changes have taken place to `.clear()`

Reply:
considering there are only 3 files modified in this PR, 2 of them are `package.json` and its lock file, and the third one being the `experimentalMemoryManagement` change to the config file to try to fix the failing tests, I think it's safe to say that the failures are due to the Cypress update...
If we want to continue updating Cypress, I think we need to re-work these 3 tests  a bit to make them work with the newer version.

Reply:
This version works. Although we're not doing like the previous test, we are just opening the combo box and selecting the element.
``` 
it('with the correct environment when changing the environment', () => {
      cy.intercept('GET', '/internal/apm/suggestions?*').as('environmentSuggestionOptions');
      cy.wait(mainAliasNames);
      cy.wait('@environmentSuggestionOptions');

      cy.getByTestSubj('environmentFilter').click();
      cy.contains('button', 'production').click();

      cy.expectAPIsToHaveBeenCalledWith({
        apisIntercepted: mainAliasNames,
        value: 'environment=production',
      });
    });
```

Reply:
Hey @michaelolo24 I was trying to fix the tests but the browser keeps crasing all the time with this message:
```
We detected that the Chrome Renderer process just crashed.

We have failed the current spec but will continue running the next spec.

This can happen for a number of different reasons.

If you're running lots of tests on a memory intense application.
  - Try increasing the CPU/memory on the machine you're running on.
  - Try enabling experimentalMemoryManagement in your config file.
  - Try lowering numTestsKeptInMemory in your config file during 'cypress open'.

You can learn more here:

https://on.cypress.io/renderer-process-crashed
```

<img width="1494" alt="Screenshot 2024-11-19 at 14 27 51" src="https://github.com/user-attachments/assets/ea5ce75c-e8ec-41ae-ba6e-0307d0df6ef2">

Unfortunatly I don't feel comfortable approving this PR until this is fixed.

Reply:
Yea, this goes back to the error I was seeing when interacting with the `environmentFilter` dropdown. Suspecting cypress crashes due to an infinite loop of trying to delete the field values, but not being able to delete...but not 100% sure. I was able to get the changes I made working locally, but because of the way the component works, it's hard to properly test the suggestion api since you can't clear the environment field. We'll see if this current ci run passes. If so, please feel free to make modifications you'd like to make to my changes to better fit APM's expectations as the tests _shouldn't_ crash any longer 🤞🏾